src/plugins/safeSnap/index.ts

```diff
@@ -135,7 +136,34 @@ export default class Plugin {
     return getModuleDetails(provider, network, moduleAddress);
   }
 
-  async *submitProposalWithHashes(
+  async submitProposalWithHashes(
+    web3: any,
+    moduleAddress: string,
+    moduleType: 'reality' | 'uma',
+    proposalId: string,
+    txHashes: string[]
+  ) {
+    switch (moduleType) {
+      case 'reality':
+        return this.submitRealityProposalWithHashes(
+          web3,
+          moduleAddress,
+          proposalId,
+          txHashes
+        );
+      case 'uma':
+        return this.submitUmaProposalWithHashes(
+          web3,
+          moduleAddress,
+          proposalId,
+          txHashes
+        );
+      default:
+        throw new Error('Unknown module type');
+    }
```

Comment:
This separation should probably happen in the `HandleOutcome` component's `submitProposal` function, because it needs the transactions and not their hashes.